### PR TITLE
Da/io1 3240 operator fee change flow

### PR DESCRIPTION
**Pull request by axelrod-blox**

### CODE REVIEW

File: src/app/components/applications/SSV/MyAccount/components/EditFeeFlow/UpdateFee/components/index.styles.ts [hunk 1]
```diff
@@ -304,6 +311,9 @@ export const useStyles = makeStyles((theme: Theme) => ({
             return `1px solid ${theme.colors.tint90}`;
         },
         color: (props: any) => {
+            if (props.step === 5) {
```
Comment (meshin-blox) [inline src/app/components/applications/SSV/MyAccount/components/EditFeeFlow/UpdateFee/components/index.styles.ts:314]: Do not use constants, create enum with steps and use them in all the places

**Review by meshin-blox — CHANGES_REQUESTED** [review]

File: src/app/components/applications/SSV/MyAccount/components/EditFeeFlow/UpdateFee/components/CancelFee.tsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+import OperatorStore from '~app/common/stores/applications/SsvWeb/Operator.store';
+import ChangeFeeDisplayValues from '~app/components/common/FeeUpdateTo/ChangeFeeDisplayValues';
+import ReactStepper
+    from '~app/components/applications/SSV/MyAccount/components/EditFeeFlow/UpdateFee/components/Stepper';
+import { useStyles } from '~app/components/applications/SSV/MyAccount/components/EditFeeFlow/UpdateFee/components/index.styles';
+import {
+    IncreaseFlowProps,
+} from '~app/components/applications/SSV/MyAccount/components/EditFeeFlow/UpdateFee/components/IncreaseFlow';
+
+const CancelFee = ({ oldFee, newFee, currentCurrency, declareNewFeeHandler } : IncreaseFlowProps) => {
```
[the remark this answers is not in the document]
Comment (axelrod-blox) [reply src/app/components/applications/SSV/MyAccount/components/EditFeeFlow/UpdateFee/components/CancelFee.tsx:17]: IncreaseFlowProps type use only in increase flow

File: src/app/components/applications/SSV/MyAccount/components/EditFeeFlow/UpdateFee/components/ChangeFee.tsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import Grid from '@mui/material/Grid';
+import Typography from '@mui/material/Typography';
+import BorderScreen from '~app/components/common/BorderScreen';
+import ConversionInput from '~app/components/common/ConversionInput/ConversionInput';
+import PrimaryButton from '~app/components/common/Button/PrimaryButton/PrimaryButton';
+import { UpdateFeeProps } from '~app/components/applications/SSV/MyAccount/components/EditFeeFlow/UpdateFee/UpdateFee';
+import { useStyles } from '~app/components/applications/SSV/MyAccount/components/EditFeeFlow/UpdateFee/components/index.styles';
+
+const ChangeFee = ({ newFee, onChangeHandler = () => null, error, nextIsDisabled, onNextHandler, setCurrency = () => null }: UpdateFeeProps) => {
```
[the remark this answers is not in the document]
Comment (axelrod-blox) [reply src/app/components/applications/SSV/MyAccount/components/EditFeeFlow/UpdateFee/components/ChangeFee.tsx:10]: They are not the same

File: src/app/components/applications/SSV/MyAccount/components/EditFeeFlow/UpdateFee/UpdateFee.tsx [hunk 4]
```diff
@@ -1,102 +1,154 @@
...
-        if (savedOperator && savedOperator?.includes(operatorStore.processOperatorId)) {
-          navigate(ssvRoutes.MY_ACCOUNT.OPERATOR.UPDATE_FEE.START);
-        } else {
-          navigate(ssvRoutes.MY_ACCOUNT.OPERATOR.UPDATE_FEE.EXPIRED);
-        }
-      }
+  const onInputChange = ( e : any ) => {
+    const { value } = e.target;
+    setNewFee(value);
+    if (value !== '') {
```
[the remark this answers is not in the document]
Comment (axelrod-blox) [reply src/app/components/applications/SSV/MyAccount/components/EditFeeFlow/UpdateFee/UpdateFee.tsx:114]: done

File: src/app/common/stores/applications/SsvWeb/Operator.store.ts [hunk 5]
```diff
@@ -12,6 +11,7 @@ import ApplicationStore from '~app/common/stores/Abstracts/Application';
 import SsvStore from '~app/common/stores/applications/SsvWeb/SSV.store';
 import MyAccountStore from '~app/common/stores/applications/SsvWeb/MyAccount.store';
 import NotificationsStore from '~app/common/stores/applications/SsvWeb/Notifications.store';
+import Operator from '~lib/api/Operator';
```
[the remark this answers is not in the document]
Comment (axelrod-blox) [reply src/app/common/stores/applications/SsvWeb/Operator.store.ts:14]: done

File: src/app/common/stores/applications/SsvWeb/Operator.store.ts [hunk 6]
```diff
@@ -493,6 +499,76 @@ class OperatorStore extends BaseStore {
...
+            })
+            .on('error', (error: any) => {
+              console.debug('Contract Error', error.message);
+              applicationStore.setIsLoading(false);
+              applicationStore.showTransactionPendingPopUp(false);
+              resolve(false);
+            });
+      } catch (e: any) {
+        console.log('<<<<<<<<<<<<<<error>>>>>>>>>>>>>>');
+        console.log(e.message);
```
[the remark this answers is not in the document]
Comment (axelrod-blox) [reply src/app/common/stores/applications/SsvWeb/Operator.store.ts]: done

File: src/app/common/stores/applications/SsvWeb/Operator.store.ts [hunk 7]
```diff
@@ -237,11 +244,18 @@ class OperatorStore extends BaseStore {
...
+      this.operatorCurrentFee = await contract.methods.getOperatorFee(operatorId).call();
+      const response = await contract.methods.getOperatorDeclaredFee(operatorId).call();
+      this.operatorFutureFee = response['0'] === '0' ? null : response['0'];
+      this.operatorApprovalBeginTime = response['1'] === '1' ? null : response['1'];
+      this.operatorApprovalEndTime = response['2'] === '2' ? null : response['2'];
+    } catch (e: any) {
+      console.error(e.message);
+      this.operatorFutureFee =  null;
+      this.operatorApprovalBeginTime = null;
+      this.operatorApprovalEndTime = null;
```
[the remark this answers is not in the document]
Comment (axelrod-blox) [reply src/app/common/stores/applications/SsvWeb/Operator.store.ts]: done

File: src/app/common/stores/applications/SsvWeb/Operator.store.ts [hunk 8]
```diff
@@ -164,6 +165,12 @@ class OperatorStore extends BaseStore {
       });
   }
 
+  refreshOperatorFeeData() {
```
[the remark this answers is not in the document]
Comment (axelrod-blox) [reply src/app/common/stores/applications/SsvWeb/Operator.store.ts]: done

File: src/app/components/applications/SSV/MyAccount/components/EditFeeFlow/UpdateFee/UpdateFee.tsx [hunk 9]
```diff
@@ -1,102 +1,154 @@
...
+    }
+  }, [newFee, error]);
+
+  const declareNewFeeHandler = () => {
+    setCurrentFlowStep(FeeUpdateSteps.START);
+  };
 
-      // @ts-ignore
-      const daysFromEndPendingStateTime = Math.ceil(Math.abs(todayDate - endPendingStateTime) / (1000 * 3600 * 24));
+  const setErrorHandler = ( errorResponse: ErrorType ) => {
```
[the remark this answers is not in the document]
Comment (axelrod-blox) [reply src/app/components/applications/SSV/MyAccount/components/EditFeeFlow/UpdateFee/UpdateFee.tsx:96]: done

File: src/app/components/applications/SSV/MyAccount/components/EditFeeFlow/UpdateFee/components/CancelFee.tsx [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+    const stores = useStores();
+    const classes = useStyles({ step: 5 });
+    const operatorStore: OperatorStore = stores.Operator;
+
+    const declareNewFee = () => {
+        operatorStore.refreshOperatorFeeData();
+        declareNewFeeHandler();
+
```
[the remark this answers is not in the document]
Comment (axelrod-blox) [reply src/app/components/applications/SSV/MyAccount/components/EditFeeFlow/UpdateFee/components/CancelFee.tsx]: done

File: src/app/components/applications/SSV/MyAccount/components/EditFeeFlow/UpdateFee/components/index.styles.ts (hunk 1 above)
Comment (axelrod-blox) [reply src/app/components/applications/SSV/MyAccount/components/EditFeeFlow/UpdateFee/components/index.styles.ts]: done

File: src/app/common/stores/applications/SsvWeb/Operator.store.ts [hunk 11]
```diff
@@ -237,11 +244,18 @@ class OperatorStore extends BaseStore {
   async getOperatorFeeInfo(operatorId: number) {
```
[the remark this answers is not in the document]
Comment (axelrod-blox) [reply src/app/common/stores/applications/SsvWeb/Operator.store.ts]: done

File: src/app/common/stores/applications/SsvWeb/Operator.store.ts [hunk 12]
```diff
@@ -493,6 +499,76 @@ class OperatorStore extends BaseStore {
...
+                new Decimal(newFee).dividedBy(config.GLOBAL_VARIABLE.BLOCKS_PER_YEAR).toFixed().toString(),
+            ),
+        );
+        await this.getOperatorFeeInfo(operatorId);
+        let operatorBefore = await Operator.getInstance().getOperator(operatorId);
+        operatorBefore = {
+          id: operatorBefore.id,
+          fee: operatorBefore.fee,
+        };
+        await contractInstance.methods.reduceOperatorFee(operatorId, formattedFee).send({ from: walletStore.accountAddress })
```
[the remark this answers is not in the document]
Comment (axelrod-blox) [reply src/app/common/stores/applications/SsvWeb/Operator.store.ts:515]: done

File: src/app/common/stores/applications/SsvWeb/Operator.store.ts [hunk 13]
```diff
@@ -493,6 +499,76 @@ class OperatorStore extends BaseStore {
...
+                    await this.refreshOperatorsAndClusters(resolve, true);
+                    break;
+                  }
+                  iterations += 1;
+                  const changed = await myAccountStore.checkEntityChangedInAccount(
+                      async () => {
+                        const operatorAfter = await Operator.getInstance().getOperator(operatorId);
+                        await this.getOperatorFeeInfo(operatorId);
+                        return {
+                          id: operatorAfter.id,
```
[the remark this answers is not in the document]
Comment (axelrod-blox) [reply src/app/common/stores/applications/SsvWeb/Operator.store.ts:531]: Here I can't do because I use same way before and get error

**Review by meshin-blox — APPROVED** [review]

**Pull request merged**